lib/semantics/expression.cc

```diff
@@ -1577,12 +1585,28 @@ static void AddPassArg(ActualArguments &actuals, Expr<SomeDerived> &&expr,
   }
   ActualArgument passed{AsGenericExpr(std::move(expr))};
   passed.set_isPassedObject(isPassedObject);
-  if (iter == actuals.end() && passName) {
-    passed.set_keyword(*passName);
+  if (iter == actuals.end()) {
+    if (auto passName{GetPassName(component)}) {
+      passed.set_keyword(*passName);
+    }
   }
   actuals.emplace(iter, std::move(passed));
 }
 
+// Return the compile-time resolution of a procedure binding, if possible.
+static const Symbol *GetBindingResolution(
+    const std::optional<DynamicType> &baseType, const Symbol &component) {
+  const auto *binding{component.detailsIf<semantics::ProcBindingDetails>()};
+  if (!binding) {
```

Comment:
Consider structuring this differently; the test on `binding` could be part of an `if` statement that declares it.